doc/developer-guides/hld/hv-partitionmode.rst

```diff
@@ -0,0 +1,358 @@
+   :align: center
+
+
+DMA
+---
+
+ACRN developers need to statically define the pass-thru devices for each
+guest using the guest configuration. For devices to DMA to/from guest
+memory directly, ACRN parses thru the list of pass-thru devices for each
+guest and creates context entries in the VT-d remapping hardware. EPT
+page tables created for the guest are used for VT-d page tables.
+
+I/O
+---
+
+ACRN supports I/O for pass-thru devices with two restrictions.
+
+1) Supports only MMIO. So requires developers to expose I/O BARs as
+   not present in the guest configuration.
+
+2) Supports only 32-bit MMIO BAR type.
+
+As guest PCI sub-system scans the PCI bus and assigns Guest Physical
+Address (GPA) to the MMIO BAR, ACRN maps GPA to the address in the
+physical BAR of the pass-thru device using EPT. Following timeline chart
+explains how PCI devices are assigned to guest and BARs are mapped upon
+guest initialization.
+
+.. figure:: images/partition-image13.png
+   :align: center
+
+
+Interrupt Configuration
+-----------------------
+
+ACRN supports both legacy (INTx) and MSI interrupts for pass-thru
+devices.
+
+INTx support
+~~~~~~~~~~~~
+
+ACRN expects developers to identify the interrupt line info (0x3CH) from
+the physical BAR of the pass-thru device and build an interrupt entry in
+the mptable for the corresponding guest. As guest configures the vIOAPIC
+for the interrupt RTE, ACRN writes the info from the guest RTE into the
+physical IOAPIC RTE. Upon guest kernel request to mask the interrupt,
+ACRN writes to the physical RTE to mask the interrupt at the physical
+IOAPIC. When guest masks the RTE in vIOAPIC, ACRN masks the interrupt
+RTE in the physical IOAPIC. Level triggered interrupts are not
+supported.
+
+MSI support
+~~~~~~~~~~~
+
+Guest reads/writes to PCI configuration space for configuring MSI
+interrupts using address. Data and control registers are pass-thru to
+the physical BAR of pass-thru device. Refer to `Configuration
+space access`_ for details on how PCI configuration space is emulated.
+
+Virtual device support
+======================
+
+ACRN provides read-only vRTC support for partition mode guests. Writes
+to the data port are discarded.
+
+For port I/O to ports other than vPIC, vRTC, or vUART, reads return 0xFF and
+writes are discarded.
+
+Interrupt delivery
+==================
+
+Guests w/o LAPIC pass-thru
+--------------------------
+
+In partition mode of ACRN, interrupts stay disabled after a vmexit.  The
+processor does not take interrupts when it is executing in VMX root
+mode. ACRN configures the processor to take vmexit upon external
+interrupt if the processor is executing in VMX non-root mode. Upon an
+external interrupt, after sending EOI to the physical LAPIC, ACRN
+injects the vector into the vLAPIC of vCPU currently running on the
+processor. Guests using Linux as kernel, uses vectors less than 0xECh
+for device interrupts.
+
+.. figure:: images/partition-image20.png
```

Comment:
@dbkinder should we use graphviz instead here?

Reply:
> @gvancuts Let's get this approved and merged, then I'll come back around and convert images to graphviz as appropriate.

Works for me, I'll remove all comments I made about graphviz so we concentrate and close on the other comments.